libraries/AP_RangeFinder/AP_RangeFinder_VL53L1X.cpp

```diff
@@ -0,0 +1,474 @@
+    write_register(GPIO__TIO_HV_STATUS, 0x02);
+    write_register(SIGMA_ESTIMATOR__EFFECTIVE_PULSE_WIDTH_NS, 8); // tuning parm default
+    write_register(SIGMA_ESTIMATOR__EFFECTIVE_AMBIENT_WIDTH_NS, 16); // tuning parm default
+    write_register(ALGO__CROSSTALK_COMPENSATION_VALID_HEIGHT_MM, 0x01);
+    write_register(ALGO__RANGE_IGNORE_VALID_HEIGHT_MM, 0xFF);
+    write_register(ALGO__RANGE_MIN_CLIP, 0); // tuning parm default
+    write_register(ALGO__CONSISTENCY_CHECK__TOLERANCE, 2); // tuning parm default
+
+    // general config
+    write_register16(SYSTEM__THRESH_RATE_HIGH, 0x0000);
+    write_register16(SYSTEM__THRESH_RATE_LOW, 0x0000);
+    write_register(DSS_CONFIG__APERTURE_ATTENUATION, 0x38);
+
+    // timing config
+    // most of these settings will be determined later by distance and timing
+    // budget configuration
+    write_register16(RANGE_CONFIG__SIGMA_THRESH, 360); // tuning parm default
+    write_register16(RANGE_CONFIG__MIN_COUNT_RATE_RTN_LIMIT_MCPS, 192); // tuning parm default
+
+    // dynamic config
+    write_register(SYSTEM__GROUPED_PARAMETER_HOLD_0, 0x01);
+    write_register(SYSTEM__GROUPED_PARAMETER_HOLD_1, 0x01);
+    write_register(SD_CONFIG__QUANTIFIER, 2); // tuning parm default
+
+    // from VL53L1_preset_mode_timed_ranging_*
+    // GPH is 0 after reset, but writing GPH0 and GPH1 above seem to set GPH to 1,
+    // and things don't seem to work if we don't set GPH back to 0 (which the API
+    // does here).
+    write_register(SYSTEM__GROUPED_PARAMETER_HOLD, 0x00);
+    write_register(SYSTEM__SEED_CONFIG, 1); // tuning parm default
+
+    // from VL53L1_config_low_power_auto_mode
+    write_register(SYSTEM__SEQUENCE_CONFIG, 0x8B); // VHV, PHASECAL, DSS1, RANGE
+    write_register16(DSS_CONFIG__MANUAL_EFFECTIVE_SPADS_SELECT, 200 << 8);
+    write_register(DSS_CONFIG__ROI_MODE_CONTROL, 2); // REQUESTED_EFFFECTIVE_SPADS
+
+    // default to long range, 50 ms timing budget
+    // note that this is different than what the API defaults to
+    setDistanceMode(Long);
+    setMeasurementTimingBudget(50000);
+
+    // the API triggers this change in VL53L1_init_and_start_range() once a
+    // measurement is started; assumes MM1 and MM2 are disabled
+    write_register16(ALGO__PART_TO_PART_RANGE_OFFSET_MM,
+                     read_register16(MM_CONFIG__OUTER_OFFSET_MM) * 4);
+
+    startContinuous(50);
+
+    // call timer() every 50ms. We expect new data to be available every 50ms
+    dev->register_periodic_callback(50000,
+                                    FUNCTOR_BIND_MEMBER(&AP_RangeFinder_VL53L1X::timer, void));
+    return true;
+}
+
+// set distance mode to Short, Medium, or Long
+// based on VL53L1_SetDistanceMode()
+bool AP_RangeFinder_VL53L1X::setDistanceMode(DistanceMode mode)
+{
+    // save existing timing budget
+    uint32_t budget_us = getMeasurementTimingBudget();
+
+    switch (mode)
```

Comment:
If I can set the mode with the config parameter, I can choose it in the situation.